### PR TITLE
UI/remove kv2 secret create or update

### PR DESCRIPTION
**Pull request by Monkeychip**
Focused on removing all kv v2 logic from the following:
**Routes:**
`secretEdit`
`list`
**Components:**
`secretEdit`
`secretCreateOrUpdate`
`secretEditToolbar`
`secretFormShow`

It was tempting to refactor a lot of this (ex: how we do validations in secret-edit), but holding off refactoring for the refactoring tickets coming later. Keeping the scope to only removal will make refactoring easier, and help us keep track of where regression bugs happened, if they happen.

It's hard to follow what is okay to remove and what is not. I focused on removing anything regarding **version** (e.g. kv v2 or the query param version) as none of the remaining engines that use this flow have the concept of version. **Metadata** is also not relevant anymore.

### CODE REVIEW

File: ui/app/models/secret-v2.js
Comment (Monkeychip) [inline ui/app/models/secret-v2.js:1]: this whole model should eventually be deleted, I removed this one item here because it is completely gone from the codebase now.

File: ui/app/routes/vault/cluster/secrets/backend/list.js [hunk 1]
```diff
@@ -99,9 +96,9 @@ export default Route.extend({
       aws: 'role-aws',
       // secret or secret-v2
       cubbyhole: 'secret',
```
Comment (Monkeychip) [inline ui/app/routes/vault/cluster/secrets/backend/list.js:98]: you can see more clearly now what model this flow uses and what engine uses it. The model is 'secret' and Cubbyhole, anything generic (kv1) and I believe transit still uses part of it.

File: ui/app/components/secret-create-or-update.js [hunk 2]
```diff
@@ -93,12 +82,6 @@ export default class SecretCreateOrUpdate extends Component {
         ? set(this.validationMessages, name, `${name} can't be blank.`)
         : set(this.validationMessages, name, '');
     }
-    // check duplicate on path
-    if (name === 'path' && value) {
-      this.secretPaths?.includes(value)
-        ? set(this.validationMessages, name, `A secret with this ${name} already exists.`)
-        : set(this.validationMessages, name, '');
-    }
```
Comment (zofskeez) [inline ui/app/components/secret-create-or-update.js:101]: Just want to double check that this is KV only? Seems like all secret types cannot have a path with the same name.

Comment (zofskeez) [inline ui/app/components/secret-create-or-update.js:101]: Just want to double check that this is KV only? Seems like all secret types cannot have a path with the same name.

Comment (Monkeychip) [reply ui/app/components/secret-create-or-update.js:101]: Good catch. This was originally a nice to have for kv v2. The behind the scenes of this is pretty gnarly as it queries every single secret, e.g. thousands, if you have thousands. 🙈 We can rely on the backend to throw a API error instead.

**Review by hellobontempo — APPROVED** [review]
Nice job removing the kv v2 logic!
